libs/http-service/src/http-service.cpp

```diff
@@ -108,14 +122,15 @@ struct HttpService::Impl
             self_.request(request);
```

Comment:
Let us change the logic here to ensure that...
1. Every LayerTilesRequest with a bad layer gets the NoDataSource status.
2. HttpService does not need to call canProcess explicitly.

To do this, we introduce `bool mapget::Service::request(vector<LayerTilesRequest::Ptr>)`. This will internally do the following:
1. Call canProcess for every LayerTilesRequest.
2. Set the status for any un-processable requests to NoDataSource.
3. If any un-processable request was found, set the status for the others to Aborted.
4. Return false if un-processable requests were found, otherwise start the requests and return true.

The HttpService evaluates the return value. For true (success), it sets the content provider. For false, it assembles a JSON response structure:

```
{
   requestStatuses: [<status of every request in the received order>]
}
```

This JSON is then sent as the response body, along with code 400. The `HttpClient` side needs to then forward these status values to the C++ request objects.